tests/unit/charms/tls_certificates_interface/v2/test_tls_certificates_v2_provides.py

```diff
@@ -364,6 +489,62 @@ def test_given_some_certificates_in_relation_data_when_set_relation_certificate_
         loaded_relation_data = _load_relation_data(dict(provider_relation_data))
         self.assertEqual(expected_relation_data, loaded_relation_data)
 
+    def test_given_some_certificates_in_relation_data_when_set_relation_certificate_and_unit_not_leader_then_certificate_is_not_added_to_relation_data(  # noqa: E501
+        self,
+    ):
+        relation_id = self.create_certificates_relation_with_1_remote_unit()
+        self.harness.set_leader(is_leader=True)
+        initial_certificate = "whatever initial cert"
+        initial_certificate_signing_request = "whatever initial csr"
+        initial_ca = "whatever initial ca"
+        initial_chain = ["whatever initial cert 1", "whatever initial cert 2"]
+        new_ca = "whatever new ca"
+        new_certificate = "whatever new certificate"
+        new_certificate_signing_request = "whatever new certificate signing request"
+        new_chain = ["whatever new cert 1", "whatever new cert 2"]
+
+        key_values = {
+            "certificates": json.dumps(
+                [
+                    {
+                        "certificate": initial_certificate,
+                        "certificate_signing_request": initial_certificate_signing_request,
+                        "ca": initial_ca,
+                        "chain": initial_chain,
+                    }
+                ]
+            )
+        }
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=self.harness.charm.app.name, key_values=key_values
+        )
+        self.harness.set_leader(is_leader=False)
+        self.harness.charm.certificates.set_relation_certificate(
+            certificate=new_certificate,
+            ca=new_ca,
+            chain=new_chain,
+            certificate_signing_request=new_certificate_signing_request,
+            relation_id=relation_id,
+        )
+
+        expected_relation_data = {
+            "certificates": [
+                {
+                    "certificate": initial_certificate,
+                    "certificate_signing_request": initial_certificate_signing_request,
+                    "ca": initial_ca,
+                    "chain": initial_chain,
+                },
+            ]
+        }
```

Comment:
This would make the test a bit more maintainable.
```suggestion
        expected_relation_data = {
            "certificates": json.loads(key_values["certificates"])
        }
```

Reply:
Done